src/NinjaTools.FluentMockServer.API/Proxy/Evaluation/Visitors/RequestEvaluatorVistor.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            if (_evaluationContext.EnsureNotNull(HttpRequest.Cookies, cookies) is {} httpCookies)
+            {
+                if (cookies.Except(httpCookies).Any())
+                {
+                    _evaluationContext.Fail(httpCookies, cookies);
+                }
+                else
+                {
+                    _evaluationContext.Match(httpCookies, cookies);
+                }
+            }
+        }
+
+
+
+        /// <inheritdoc />
+        public void VisitHeaders(IDictionary<string, string[]>? headers)
+        {
+            if (_evaluationContext.EnsureNotNull(HttpRequest.Headers, headers) is {} httpRequestMember)
+            {
+                if (headers.Except(httpRequestMember?.ToDictionary(k => k.Key, v => v.Value.ToArray())).Any())
+                {
+                    _evaluationContext.Fail(httpRequestMember, headers);
+                }
+                else
+                {
+                    _evaluationContext.Match(httpRequestMember, headers);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void VisitPath(string? path)
+        {
+            if (_evaluationContext.EnsureNotNull(HttpRequest.Path.Value, path) is {} httpPath)
+            {
+                if(httpPath != path)
+                {
+                    _evaluationContext.Fail(httpPath, path);
+                }
+                else
+                {
+                    _evaluationContext.Match(httpPath, path);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void VisitMethod(string? method)
+        {
+            if (_evaluationContext.EnsureNotNull(HttpRequest.Method, method) is {} httpMethod)
+            {
+                if (httpMethod != method)
+                {
+                    _evaluationContext.Fail(httpMethod, method);
+                }
+                else
+                {
+                    _evaluationContext.Match(httpMethod, method);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void VisitQuery(string? query)
+        {
+            if (_evaluationContext.EnsureNotNull(HttpRequest.QueryString.Value, query) is {} httpQuery)
+            {
+                if (httpQuery != query)
+                {
+                    _evaluationContext.Fail(httpQuery, query);
+                }
+                else
+                {
+                    _evaluationContext.Match(httpQuery, query);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void VisitBody(RequestBodyMatcher? bodyMatcher)
+        {
+            if (bodyMatcher != null)
+            {
+                VisitBody(bodyMatcher.Content, bodyMatcher.MatchExact, bodyMatcher.Type);
+            }
+        }
+
+        /// <inheritdoc />
+        private void VisitBody(string? requestBody, bool exactMatch, RequestBodyKind kind)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'kind'.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)